utils/pass-generator.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+const config = require('../config');
+const logger = require('hof/lib/logger')({ env: config.env });
+
+
+const isPasswordPolicy = value => {
+  (!value.match(/^(?=.*[A-Z])(?=.*[a-z])/,
+    /(?=.*[!#$%&*+=?@])/,
+    /(?=.*\d)[A-Za-z!#$%&*+=?@0-9]{8,}$/)) ? false : true;
+};
+
+const generatePassword =  async (length, characterType) => {
+  let randomPassword = '';
+  let isPolicyMatch;
+  try {
+    const cryptoRandomString = await import('crypto-random-string');
+    do{
+      randomPassword = cryptoRandomString.default({length: length, characters: characterType});
+      isPolicyMatch = isPasswordPolicy(randomPassword);
+    }
+    while(isPolicyMatch);
+  }catch(errorMessage) {
+    logger.log('error', `error generating password: ${errorMessage}`);
+  }
```

Comment:
Need to throw error here as well
```suggestion
  } catch (error) {
    const errorMessage = `Failed to generate password: ${errorMessage}`;
    logger.log('error', errorMessage);
    throw new Error(errorMessage);
  }
  ```